### PR TITLE
Update netpol tests to use e2e framework

### LINKED ISSUE #108298: test/e2e: use framework helpers to create namespaces
### What would you like to be added?

Currently, some e2e tests do not use central framework helper methods to create namespaces. Examples:

https://github.com/kubernetes/kubernetes/blob/296bf4f01668374ade252a751d4c3567917b9890/test/e2e/apimachinery/webhook.go#L1532
https://github.com/kubernetes/kubernetes/blob/296bf4f01668374ade252a751d4c3567917b9890/test/e2e/apimachinery/webhook.go#L2370-L2375
https://github.com/kubernetes/kubernetes/blob/296bf4f01668374ade252a751d4c3567917b9890/test/e2e/network/netpol/kubemanager.go#L178

The fix is to use https://github.com/kubernetes/kubernetes/blob/296bf4f01668374ade252a751d4c3567917b9890/test/e2e/framework/framework.go#L519 instead.

### Why is this needed?

This makes maintanance of e2e tests hard as enforcement settings (like pod security admission) is not centralized

### PR DESCRIPTION
**Pull request by astoycos**
#### What type of PR is this?

/kind bug

Update the newer netpol test suite to utilize
the built in e2e framework. This should fix problems
related to serviceAccount creation delays and pod
security policies.

This is causing issues downstream in OCP see https://github.com/openshift/origin/pull/26775. 

Specifically this PR fixes the sig-network related e2e tests, but there are still 
more e2e tests that do this incorrectly which will need to be fixed in follow up PRs

Works to Fix #108298

#### Special notes for your reviewer:

There is one other palace we don't use a framework for NS creation  in the networking domain 

`test/e2e/network/scale/localrun/ingress_scale.go` however it is an old test designed for local testing 
and IMO does not need to be touched. (Please let me know if this assumption is incorrect.)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/network/netpol/kubemanager.go [hunk 1]
```diff
@@ -66,7 +65,7 @@ func newKubeManager(framework *framework.Framework) *kubeManager {
 func (k *kubeManager) initializeCluster(model *Model) error {
 	var createdPods []*v1.Pod
 	for _, ns := range model.Namespaces {
-		_, err := k.createNamespace(ns.Spec())
+		_, err := k.framework.CreateNamespace(ns.Spec().Name, map[string]string{})
```
Comment (stlaz) [inline test/e2e/network/netpol/kubemanager.go:68]: This omits the label selector of `ns.Spec()` of the original NS creation, I wonder whether this was intended.

Also, a small nit: you can do just `ns.Name`.

All and all, I think the following might work a little better:
```golang
_, err := k.framework.CreateNamespace(ns.Name, ns.LabelSelector())
```

Comment (astoycos) [reply test/e2e/network/netpol/kubemanager.go]: Ack, Done!

**Comment by astoycos** [comment]
Closed in favor of https://github.com/kubernetes/kubernetes/pull/111789

**Pull request closed without merging**